samples/Rewrite/Condition/ConditionExpression.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Text.RegularExpressions;
+
+namespace Rewrite.ConditionParser
+{
+    public class ConditionExpression : Expression
+    {
+        public Operand Operand { get; set; }
+        public bool Invert { get; set; }
+       
+        public bool VisitConditionExpression(HttpContext context, Match results, Match previous, string concatTestString)
```

Comment:
this should not be a method on the expression. the visitor with all the methods is a separate class. The base visitor has has all the methods virtual and they simply go throught the graph. That way, you can implement your own visitor by overiding parts of the original visitor without having the implement everything and it also allows you to stop the visitor if you don't need to visit the child of one node